internal/wrappers/vorpal/scans/vorpal-grpc.go

```diff
@@ -0,0 +1,117 @@
+package scans
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/checkmarx/ast-cli/internal/logger"
+	"github.com/checkmarx/ast-cli/internal/wrappers/vorpal/scans/protos"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+)
+
+type VorpalWrapper struct {
+	port int
+}
+
+var (
+	vorpalScanErrMsg = "Vorpal scan failed for file %s"
+	grpcConnErrMsg   = "Error occurred while creating the grpc client in address %q. error: %v"
+)
+
+func NewVorpalWrapper(port int) *VorpalWrapper {
+	return &VorpalWrapper{
+		port: port,
+	}
+}
+
+func (s *VorpalWrapper) Scan(filePath, sourceCode string) (*protos.ScanResult, error) {
+	options := []grpc.DialOption{
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithBlock(),
+	}
+
+	timeout := 1 * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	localhostAddress := fmt.Sprintf("0.0.0.0:%d", s.port)
+	conn, err := grpc.DialContext(ctx, localhostAddress, options...)
+	defer func(conn *grpc.ClientConn) {
+		_ = conn.Close()
+	}(conn)
+
+	if err != nil {
+		logger.Printf("grpc.DialContext(%q): %v", localhostAddress, err)
+		return nil, errors.Wrapf(err, vorpalScanErrMsg, filePath)
+	}
+
+	client := protos.NewScanServiceClient(conn)
+
+	request := &protos.SingleScanRequest{
+		ScanRequest: &protos.ScanRequest{
+			Id:         uuid.New().String(),
+			FileName:   filePath,
+			SourceCode: sourceCode,
+		},
+	}
+
+	scanResultResponse, err := client.Scan(ctx, request)
+	if err != nil {
+		return nil, errors.Wrapf(err, vorpalScanErrMsg, filePath)
+	}
+
+	return scanResultResponse, nil
+}
+
+func checkHealth(service string, conn grpc.ClientConnInterface) (*healthpb.HealthCheckResponse, error) {
+	req := &healthpb.HealthCheckRequest{
+		Service: service,
+	}
+	ctx := context.Background()
+	rsp, err := healthpb.NewHealthClient(conn).Check(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	return rsp, nil
+}
+
+func (s *VorpalWrapper) CheckHealth() error {
+	timeout := 1 * time.Second
+	options := []grpc.DialOption{
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithBlock(),
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	localhostAddress := fmt.Sprintf("0.0.0.0:%d", s.port)
+	conn, err := grpc.DialContext(ctx, localhostAddress, options...)
+	if err != nil {
+		logger.Printf(grpcConnErrMsg, localhostAddress, err)
+		return err
+	}
+
+	defer func(conn *grpc.ClientConn) {
+		if conn != nil {
+			_ = conn.Close()
+		}
+	}(conn)
+
+	healthRes, healthErr := checkHealth("MicroSastEngine", conn)
```

Comment:
Please change the name from micro Sast to Vorpal engine

Reply:
Note: MicroSastEngine is still the name of the service in the code.
I changed it to VorpalEngine and Joao/Hugo will change it to this name in the future